src/bin/miri.rs

```diff
@@ -192,8 +244,16 @@ fn main() {
         };
 
         // We cannot use `rustc_driver::main` as we need to adjust the CLI arguments.
-        let mut callbacks = rustc_driver::TimePassesCallbacks::default();
-        run_compiler(env::args().collect(), &mut callbacks, insert_default_args)
+        run_compiler(
+            env::args().collect(),
+            &mut MiriBeRustCompilerCalls { target_crate },
+            // Don't insert `MIRI_DEFAULT_ARGS`, in particular, `--cfg=miri`, if we are building
+            // a "host" crate. That may cause procedural macros (and probably build scripts) to
+            // depend on Miri-only symbols, such as `miri_resolve_frame`:
+            // https://github.com/rust-lang/miri/issues/1760
+            #[rustfmt::skip]
```

Comment:
What does rustfmt do here if we don't suppress it?

Reply:
```diff
diff --git a/src/bin/miri.rs b/src/bin/miri.rs
index f7aed6bc..3de3a9a6 100644
--- a/src/bin/miri.rs
+++ b/src/bin/miri.rs
@@ -251,8 +251,8 @@ fn main() {
             // a "host" crate. That may cause procedural macros (and probably build scripts) to
             // depend on Miri-only symbols, such as `miri_resolve_frame`:
             // https://github.com/rust-lang/miri/issues/1760
-            #[rustfmt::skip]
-            /* insert_default_args: */ target_crate,
+            /* insert_default_args: */
+            target_crate,
         )
     }
 ```

Reply:
Okay... yeah that's not great. The skip is also costing an extra line though, so either way works for me.^^